### PR TITLE
Added Camunda workflow integration

### PR DESCRIPTION
**Pull request by nageshl**
#121

### CODE REVIEW

File: learn/howtos-documents.md [hunk 1]
```diff
@@ -85,6 +85,7 @@ Find the how to topics and look for sub-categories under each of the specified c
 *   [How to schedule a Java Service](/learn/how-tos/scheduling-java-service/)
 *   [Custom Query Syntax to access DB Service APIs from Java Service](/learn/app-development/services/database-services/orm-artifacts/#custom-query-syntax)
 *   [How to accomplish Pre-Post Processing for a DB Service APIs](/learn/how-tos/pre-post-processing-db-service-apis/)
+*   [How to Integrate Camunda with WaveMaker Apps](/learn/how-tos/camunda-wavemaker-app-integration/)
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/howtos-documents.md]: Lets change this to How to design a workflow in Camunda BPM and integrate into a WaveMaker app

File: learn/how-tos/camunda-wavemaker-app-integration.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+1. Getting Started With Camunda
+   https://docs.camunda.org/get-started/quick-start/
+
+1. [Download and Install Camunda](https://docs.camunda.org/enterprise/download/)
+   Note: For Integrating WM online server, Please install Camunda services on cloud such as AWS/Azure.
+
+1. Create BPMN Process for Apply Leave Request
+   Below is the process flow create in Camunda.
+    [![lftr_sel](/learn/assets/leave_processflow.png)](/learn/assets/leave_processflow.png)
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/how-tos/camunda-wavemaker-app-integration.md]: leave_processflow.png has lot of whitespace on the top. This is how it looks.
image


Can you crop that out?

File: learn/how-tos/camunda-wavemaker-app-integration.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+## Overview
+
+In this article you will learn how to add workflow into a WaveMaker application by integrating with a opensource BPM product - Camunda. We will use the flexibility of the code generated by WaveMaker and call rules modelled in Camunda.
+
+For demonstration purpose, we will use Employee Leave Management application. The Leave Management application. Employee will apply for the leave using UI and the request goes to the backend where Camunda business rules engine is integrated and based on leave balance, Camunda REST API will return the list of approvers to process the request.
+
+### STEP 1 – CAMUNDA SETUP
+
+1. About Camunda
+  Camunda BPM is a light-weight, open-source platform for Business Process Management. Camunda is a Java-based framework supporting BPMN for workflow and process automation, CMMN for Case Management and DMN for Business Decision Management.
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/how-tos/camunda-wavemaker-app-integration.md]: The sentence and "About Camunda" are coming in the same line.

File: learn/how-tos/camunda-wavemaker-app-integration.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    Below is the decision table created using the Camunda DMN.
+    [![lftr_sel](/learn/assets/business_rules.png)](/learn/assets/business_rules.png)
+
+1. Create a REST End-point for Camunda WorkFlow Engine
+   You can create the Camunda REST endpoint url. [Refer to this Camunda documentation](https://docs.camunda.org/manual/7.12/reference/rest/decision-definition/get/)
+    Example: `POST /decision-definition/key/{key}`
+
+```
+    `POST http://<camunda-host>/engine-rest/decision-definition/key/approve_leave/evaluate`
+    HTTP Headers: Content-Type="application/json"
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/how-tos/camunda-wavemaker-app-integration.md]: Lets split the code here and insert a new code block

File: learn/how-tos/camunda-wavemaker-app-integration.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+            }
+        }
+    ]
+```
+
+### STEP 2 – Build Leave Management WaveMaker App
+
+In leave management application, employee will apply for the leaves entering start date & end date. Note: Initial leave balance for all the employees will be 10. Depending on the applied leave by employee Camunda workflow will process it and request will go to the respected person i.e. Manager, Director or HR. 
+	
+    Create the UI in WaveMaker for Leave Management application:
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/how-tos/camunda-wavemaker-app-integration.md]: needs fixing
image

**Comment by prasanth-reddy-wavemaker-com** [comment]
@nageshl we can make the IP address of Camunda server an application profile property, so that it can be different in each env the application deploys to. Can we mention this in the write up?

File: learn/howtos-documents.md (hunk 1 above)
Comment (nageshl) [reply learn/howtos-documents.md]: yes done.

File: learn/how-tos/camunda-wavemaker-app-integration.md (hunk 2 above)
Comment (nageshl) [reply learn/how-tos/camunda-wavemaker-app-integration.md]: Yes done.

File: learn/how-tos/camunda-wavemaker-app-integration.md (hunk 3 above)
Comment (nageshl) [reply learn/how-tos/camunda-wavemaker-app-integration.md]: Yes Done.

File: learn/how-tos/camunda-wavemaker-app-integration.md (hunk 5 above)
Comment (nageshl) [reply learn/how-tos/camunda-wavemaker-app-integration.md]: Yes done.

**Comment by nageshl** [comment]
> @nageshl we can make the IP address of Camunda server an application profile property, so that it can be different in each env the application deploys to. Can we mention this in the write up?

@prasanth-reddy-wavemaker-com 
We just need Camunda REST Endpoint host names for different env that can be mapped in WM app.

**Pull request merged**